### ISSUE TITLE
Investigate adding a handler policy to AZ::Event to support AZ::OrderedEvent

### ISSUE THREAD

**Issue by hultonha** [o3de/o3de#11146]
[Migrated from JIRA LYN-1929]
Note: This task is to refactor AZ::OrderedEvent (using a policy) to remove duplication between AZ::Event and AZ::OrderedEvent

**Comment by lumberyard-employee-dm** [comment]
It looks like the [OrderedEventHandler](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/EBus/OrderedEvent.h#L80-L125) and [EventHandler](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/EBus/Event.h#L95-L140) classes can be the same class.

**Comment by hultonha** [comment]
> It looks like the [OrderedEventHandler](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/EBus/OrderedEvent.h#L80-L125) and [EventHandler](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/EBus/Event.h#L95-L140) classes can be the same class.

Agreed - this would definitely be a useful bit of tech-debt to clean-up if someone from sig/core has time 🙂 Thanks for reviewing
